docs/builtins.md

```diff
@@ -438,3 +438,124 @@ number, but it is typically used to indicate whether the program
 terminated successfully or with an error. A status code of 0 means that
 the program terminated successfully, while any other status code is
 considered an error.
+
+## Conversion
+
+### `str2num`
+
+`str2num` converts a string to a number. If the string is not a valid
+number, it returns `0` and sets the global `err` variable to `true`.
+
+#### Example
+
+```evy
+n:num
+n = str2num "1"
+print "n:" n "err:" err
+n = str2num "NOT-A-NUMBER"
+print "n:" n "err:" err
+```
+
+Output
+```evy:output
+n: 1 err: false
+n: 0 err: true
+```
+
+#### Reference
+
+    str2num:num s:string
+
+The `str2num` function converts a string to a number. It takes a single
+argument, which is the string to convert. If the string is a valid
+number, the function returns the number. Otherwise, the function
+returns 0 and sets the global `err` variable to `true`. For more
+information on errors, see the [error section](#error).
+
+### `str2bool`
+
+`str2bool` converts a string to a boolean. If the string is not a valid
+boolean, it returns `0` and sets the global `err` variable to `true`.
+
+#### Example
+
+```evy
+b:bool
+b = str2bool "true"
+print "b:" b "err:" err
+b = str2bool "NOT-A-BOOL"
+print "b:" b "err:" err
+```
+
+Output
+```evy:output
+b: true err: false
+b: false err: true
+```
+
+#### Reference
+
+    str2bool:bool s:string
+
+The `str2bool` function converts a string to a bool. It takes a single
+argument, which is the string to convert. The function returns `true`
+if the string is equal to `"true"`, `"True"`, `"TRUE"`,  or `"1"`, and
+`false` if the string is equal to `"false"`, `"False"`, `"FALSE"`, or
+`"0"`. The function returns `false` and sets the global `err` variable
+to `true` if the string is not a valid boolean.  For more information
+on errors, see the [error section](#error).
+
+## Error
+
+The `err` variable is used to indicate whether an _non-fatal error_ has
+occurred. The `errmsg` variable contains a detailed message about the
+error that occurred.
+
+Evy programs in execution can report two types of errors:
+
+- **Fatal errors** cause the program to exit immediately with an error
+    message. These errors can typically be prevented with guarding
+    code, such as checking the array length to avoid a fatal index out
+    of bounds error.
+- **Non-fatal errors** are caused by code that could not be prevented
+    from running, such as converting a user input string to a number if
+    the string is not a number. Non-fatal errors will set the global
+    `err` variable to `true` and the program will continue executing.
+    If there is no error, `err` is set to `false`. It is therefore the
+    program's responsibility to check `err` after any possible
+    occurrence of an error.
+
+The global `errmsg` variable contains a detailed message about the error
+which is set alongside `err`. `errmsg` is set to the empty string `""`
+if no error has occurred.
```

Comment:
somewhere in this section you should state what happens to these variables when you call another function that could set them. Do they get reset if there is no error, or do they stay set if set previously and the caller should clear them before calling the function?

Reply:
ok. i tried to say this just above, but i can see this isn't clear enough. I have added a little to this paragraph and started a new paragraph describing the reset situation. please let me know if that's better.

Reply:
well, i am a bit stupid it seems. I attached the comment to exactly what i was saying was missing :facepalm: . Sorry, I just didn't see that, but also I was thinking about the `err` variable - does that get reset to false if there is no error?